pkg/policies/policies.go

```diff
@@ -85,68 +97,17 @@ func (pv *PolicyVerifier) Verify(ctx context.Context) ([]*engine.PolicyViolation
 		pv.state.Attestation.Policies = append(pv.state.Attestation.Policies, &v12.Policy{
 			Name:       spec.Metadata.Name,
 			Attachment: policyAtt,
-			Body:       string(script.Source),
 			Violations: policyViolationsToAttestationViolations(res),
 		})
 	}
 
 	return violations, nil
 }
 
-func (pv *PolicyVerifier) loadSpec(attachment *v1.PolicyAttachment) (*v1.Policy, error) {
-	// look for the referenced policy spec (note: loading by `name` is not supported yet)
-	reference := attachment.GetRef()
-	// this method understands env, http and https schemes, and defaults to file system.
-	rawData, err := blob.LoadFileOrURL(reference)
-	if err != nil {
-		return nil, fmt.Errorf("loading policy spec: %w", err)
-	}
-	jsonContent, err := crafter.LoadJSONBytes(rawData, filepath.Ext(reference))
-	if err != nil {
-		return nil, fmt.Errorf("loading policy spec: %w", err)
-	}
-	var policy v1.Policy
-	if err := protojson.Unmarshal(jsonContent, &policy); err != nil {
-		return nil, fmt.Errorf("unmarshalling policy spec: %w", err)
-	}
-	// Validate just in case
-	validator, err := protovalidate.New()
-	if err != nil {
-		return nil, fmt.Errorf("validating policy spec: %w", err)
-	}
-	err = validator.Validate(&policy)
-	if err != nil {
-		return nil, fmt.Errorf("validating policy spec: %w", err)
-	}
-
-	return &policy, nil
-}
-
-// loads a policy referenced from the spec
-func (pv *PolicyVerifier) loadPolicyScriptFromSpec(spec *v1.Policy) (*engine.Policy, error) {
-	var content []byte
-	var err error
-
-	switch source := spec.GetSpec().GetSource().(type) {
-	case *v1.PolicySpec_Embedded:
-		content = []byte(source.Embedded)
-	case *v1.PolicySpec_Path:
-		content, err = blob.LoadFileOrURL(source.Path)
-		if err != nil {
-			return nil, fmt.Errorf("loading policy content: %w", err)
-		}
-	default:
-		return nil, fmt.Errorf("policy spec is empty")
-	}
-
-	return &engine.Policy{
-		Name:   spec.GetMetadata().GetName(),
-		Source: content,
-	}, nil
-}
-
 // load the subject of the policy.
-func (pv *PolicyVerifier) loadSubject(ctx context.Context, attachment *v1.PolicyAttachment, spec *v1.Policy, state *v12.CraftingState) ([]byte, error) {
+func (pv *PolicyVerifier) loadSubject(ctx context.Context, attachment *v1.PolicyAttachment, spec *v1.Policy) ([]byte, error) {
+	state := pv.state
+
 	// Load the affected material or attestation, and checks if the expected name and type match
 	name := attachment.GetSelector().GetName()
```

Comment:
To me the name should be a filter between all the types that a policy affects to, not the differentiator between a material policy or an attestation policy. WDYT?

Reply:
I like it. We'll give priority to the type, and then filter by name if it's provided. I'll send another revision of the PR.